theano/tensor/opt.py

```diff
@@ -1454,13 +1639,8 @@ def local_fill_sink(node):
             new_client.owner.outputs[0].clients = client.outputs[0].clients
             r = local_fill_sink.transform(new_client.owner)
             if not r:
-                all_clients_replaced = False
                 continue
             replacements.update(r)
-        else:
-            all_clients_replaced = False
-    if all_clients_replaced:
-        replacements.pop(node.outputs[0], None)
```

Comment:
I timmed this again. If I add them back, it don't finisih compilation with the GSoC slow case in multiple hours. Otherwise, it finish in 471s.